src/plugins/syscalls/linux.cpp

```diff
@@ -259,7 +249,6 @@ static event_response_t linux_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
     }
 
     vmi_read_addr_va(vmi, info->regs->rsp, 0, &ret);
-    drakvuf_release_vmi(drakvuf);
```

Comment:
I see this release comes before vmi is passed to `linux_build_argbuf` down later. I would be OK with this release being moved down. I think `vmi_lock_guard` doesn't work properly if its scoped to be for the entire function, only works if its within another `{}` block inside.

Reply:
The `vmi_lock_guard` object's destructor will be called on any fuctions's body exit (by return or by throw exception).

Problems arise when a function calls another function that also captures an `vmi` lock. In this case, this does not happen.

Using of RAII `vmi_lock_guard` object allows you to avoid bugs with premature release of the lock.